src/Symfony/Component/Dsn/ConnectionFactory.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Dsn;
+
+use Symfony\Component\Dsn\Exception\InvalidArgumentException;
+use Symfony\Component\Dsn\Factory\MemcachedConnectionFactory;
+use Symfony\Component\Dsn\Factory\RedisConnectionFactory;
+
+/**
+ * Factory for undetermined Dsn.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+class ConnectionFactory
+{
+    const TYPE_REDIS = 1;
+    const TYPE_MEMCACHED = 2;
```

Comment:
if this would be logged anywhere for some reason, `2` is magic value. I advice to make value matching constant, like `MEMCACHED` here